src/components/section/ArticleList.jsx

```diff
@@ -0,0 +1,15 @@
+import { useSelector } from "react-redux";
+import NewsCard from "../ui/NewsCard";
+
+const ArticleList = () => {
+  const newsArticle = useSelector((state) => state.news.articles);
+  return (
+    <div className="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 gap-8 py-8 px-5 md:px-16 mb-10">
+      {newsArticle.articles?.slice(0, 12).map((article, index) => (
```

Comment:
no need to slice, you can order the number of items you want, instead of fetching or the data and slice them. Image if you have 1one million articles, fetch all the data to only get 10 of them is not good practice, in the api documention there  is a property called pageSize, check it.